.github/workflows/ui_test.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and run UI test
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build and run aodn portal
+        uses: addnab/docker-run-action@v3
+        with:
+          image: node:20
+          options: |
+            -v ${{ github.workspace }}:/app
+            -d -p 5173:5173 --name aodn_portal_dev
+          run: |
+            cd app
+            : # set-version will always download the latest version of yarn
+            yarn policies set-version 4.3.0
+            yarn install --immutable
+            yarn dev --host 0.0.0.0
```

Comment:
One question, how can the dev run all these test locally before push? I  believe you need to start the server and then run these test in console mode.  

However can it be more automated so that you can run yarn playright and have everything run automatically include start the docker locally, then you can use the same in the github action. So you know it works locally, then github works rather then have something diff and you need to debug from two places